src/Features/Core/Portable/GenerateMember/GenerateDefaultConstructors/AbstractGenerateDefaultConstructorsService.AbstractCodeAction.cs

```diff
@@ -72,9 +70,16 @@ protected override async Task<Document> GetChangedDocumentAsync(CancellationToke
 
             private static Accessibility DetermineAccessibility(IMethodSymbol baseConstructor, INamedTypeSymbol classType)
             {
+                // If our base is abstract, and we are not, then (since we likely want to be
+                // instantiated) we make our constructor public by default.
                 if (baseConstructor.ContainingType.IsAbstractClass() && !classType.IsAbstractClass())
                     return Accessibility.Public;
 
+                // If our base constructor is public, and we're abstract, we switch to being
+                // protected as that's a more natural default for constructors in abstract classes.
+                if (classType.IsAbstractClass() && baseConstructor.DeclaredAccessibility == Accessibility.Public)
+                    return Accessibility.Protected;
```

Comment:
this makes the constructor less accessible than it would have been before.  so this is a safe change.